compat/terminal.c

```diff
@@ -286,28 +278,36 @@ void restore_term(void)
 {
 }
 
+static int enable_non_canonical(void)
+{
+	return -1;
+}
+
 char *git_terminal_prompt(const char *prompt, int echo)
 {
 	return getpass(prompt);
 }
 
 int read_key_without_echo(struct strbuf *buf)
 {
-	static int warning_displayed;
-	const char *res;
+	BUG("should never be called if CBREAK is not supported");
+	return EOF;
+}
 
-	if (!warning_displayed) {
-		warning("reading single keystrokes not supported on this "
-			"platform; reading line instead");
-		warning_displayed = 1;
+#endif
+
+int terminal_support(enum terminal_mode mode)
```

Comment:
I had trouble understanding the purpose of this function from its name. I think something like `test_terminal_support` would be clearer.